etna/transforms/encoders/mean_encoder.py

```diff
@@ -0,0 +1,269 @@
+import reprlib
+from enum import Enum
+from typing import List
+from typing import Union
+
+import numpy as np
+import pandas as pd
+from bottleneck import nanmean
+
+from etna.datasets import TSDataset
+from etna.transforms import IrreversibleTransform
+
+
+class EncoderMode(str, Enum):
+    """Enum for different encoding strategies."""
+
+    per_segment = "per-segment"
+    macro = "macro"
+
+    @classmethod
+    def _missing_(cls, value):
+        raise ValueError(f"The strategy '{value}' doesn't exist")
+
+
+class MissingMode(str, Enum):
+    """Enum for handle missing strategies."""
+
+    category = "category"
+    global_mean = "global_mean"
+
+    @classmethod
+    def _missing_(cls, value):
+        raise NotImplementedError(
+            f"{value} is not a valid {cls.__name__}. Supported types: {', '.join([repr(m.value) for m in cls])}"
+        )
+
+
+class MeanEncoderTransform(IrreversibleTransform):
+    """
+    Makes encoding of categorical feature.
+
+    For timestamps that were seen in ``fit`` transformations are made using the formula below:
+
+    .. math::
+        \\frac{TargetSum + GlobalMean * Smoothing}{FeatureCount + Smoothing}
+    where
```

Comment:
There is smth wrong with rendering, I will come up

Reply:
Fixed, but now list items are highlighted, I don't know is it ok

Reply:
It isn't ok. I'll look into it.

Reply:
You should write:
```
    where

    * TargetSum is the sum of target up to the current timestamp for the current category, not including the current timestamp
    * RunningMean is target mean up to the current timestamp, not including the current timestamp
    * FeatureCount is the number of categories with the same value as in the current timestamp, not including the current timestamp

    For future timestamps:

    * for known categories encoding are filled with global mean of target for these categories calculated during ``fit``
    * for unknown categories encoding are filled with global mean of target in the whole dataset calculated during ``fit``

    All types of NaN values are considering as one category.
```

Reply:
Fixed